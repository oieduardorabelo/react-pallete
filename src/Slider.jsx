import React from 'react';

const Slider = (props) => (
  <input
    style={props.styles}
    name={props.name} type="range"
    min="0" max="255" step="1"
    value={props.value}
    onChange={props.onChange}
  />
)

export default Slider
