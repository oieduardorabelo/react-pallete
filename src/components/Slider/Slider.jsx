import React from 'react';

const Slider = (props) => (
  <input
    style={props.styles}
    name={props.name} type="range"
    min="0" max="255" step="1"
    value={props.value}
    onChange={props.onChange}
  />
)

Slider.propTypes = {
  styles: React.PropTypes.shape({}),
  name: React.PropTypes.string.isRequired,
  value: React.PropTypes.number.isRequired,
  onChange: React.PropTypes.func.isRequired,
}

export default Slider
