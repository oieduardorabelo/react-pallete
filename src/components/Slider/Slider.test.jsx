import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Slider from './Slider';

it('renders correctly', () => {
  const wrapper = shallow(
    <Slider name="test" value={0} onChange={jest.fn()} />
  );

  expect(toJson(wrapper)).toMatchSnapshot();
});

it('calls correct onChange spy', () => {
  const spy = jest.fn();
  const wrapper = shallow(
    <Slider name="test" value={0} onChange={spy} />
  );

  wrapper.simulate('change');

  expect(spy).toHaveBeenCalledTimes(1);
});
