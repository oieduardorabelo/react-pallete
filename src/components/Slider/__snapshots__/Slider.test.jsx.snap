exports[`test renders correctly 1`] = `
<input
  max="255"
  min="0"
  name="test"
  onChange={[Function]}
  step="1"
  type="range"
  value={0} />
`;
